File: RecyclerViewJava/app/src/main/java/com/devwilly/recyclerview_java/viewholders/BaseViewHolder.java
package com.devwilly.recyclerview_java.viewholders;

import android.support.v7.widget.RecyclerView;
import android.view.View;


/**
 * Created by Willy on 15/01/2018.
 */

public abstract class BaseViewHolder<T> extends RecyclerView.ViewHolder {

    public BaseViewHolder(View itemView) {
        super(itemView);
    }

    public abstract void bindView(T t);
}

File: RecyclerView-Kotlin/app/src/main/java/com/devwilly/recyclerviewkotlin/viewholders/BaseViewHolder.kt
package com.devwilly.recyclerviewkotlin.viewholders

import android.support.v7.widget.RecyclerView
import android.view.View

/**
 * Created by Willy on 15/01/2018.
 */
abstract class BaseViewHolder<T>(itemView: View) : RecyclerView.ViewHolder(itemView) {

    abstract fun bindView(t: T)
}